### DISCUSSION TITLE
split_ci function does not run

### DISCUSSION THREAD

**Discussion by SeraMaren** [tpronk/splithalfr#3] [Q&A] [answered]
hi!

I am calculating the splithalf reliability for a computer task. The task does not have different conditions, and included 282 trials. In the end, each trial gives one score (0= correctly identified stimulus in control location, 1 = correctly identified stimulus in experimental stimuli location, NA = not correctly identified). Looking into different vignettes for splithalfr,  I thought the RAPI sumscore (https://rdrr.io/cran/splithalfr/f/vignettes/rapi_sum.Rmd) was here most appropriate.

I went exactly by the RAPI example on the website, and everything worked except for one thing. Just when I want to calculate the CI intervals in the last step exactly as in your example: split_ci(split_scores, coefs), it gives me this error message:

Error in fn_coef(ds[, paste("score_1", split_i, sep = ".")], ds[, paste("score_2",  : Could not find "fn_coef" function

Interestingly, when I run the example code from the website with the example data, I get the same error message.
Would you know why this error may appear? Thanks!

[I am using R 4.0.3.]

**Comment by tpronk** [comment, marked as answer]
Hey,

Thanks for reporting this issue! The vignettes are indeed incorrect. The second argument to `split_ci` should be one of the functions for calculating a reliability coefficient (i.e., `spearman_brown`, `flanagan_rulon`, or `angoff_feldt`). Concretely, this means the line below should work. Note that `split_ci` is quite slow; the calculations are very intensive and the package I use for it (bcaboot) is not yet optimized to run on multiple cores.
```split_ci(split_scores, spearman_brown)```

Since you found an issue and reported it in a way that made it very easy for me to reproduce the issue and fix it, may I acknowledge you in the next release? If so, got any personal web-page that I can link to?

Cheers Thomas

**Reply by SeraMaren** [reply]
Hi Thomas,

I re-ran it now, and indeed it is working! Thanks so much for the help, and I am happy that you were able to solve them problem.

I now get this output:
$lims
                   bca             jacksd                 std      pct
0.025 0.8243435 0.0032510499 0.8339039 0.029
0.5   0.8781509 0.0007085547 0.8792684 0.462
0.975 0.9159272 0.0017391863 0.9246329 0.979

$stats
             theta          sdboot               z0                        a            sdjack
est 0.8792684 0.0231455947 -0.04764396 0.04474305 0.02215705
jsd 0.0000000 0.0007881281  0.03185005 0.00000000 0.00000000

$B.mean
[1] 1000.0000000    0.8773202

$ustats
        ustat           sdu 
0.88121659 0.02189488 

For the confidence intervals, I am looking at the bca column am I right?

Of course, I would be delighted! I don't have any personal webpage, so I think it would just be my name?

Thanks again!

**Reply by tpronk** [reply]
The documentation about split_ci can be viewed by running `?split_ci` in the console. It does make mentioning of how these CIs are calculated, but I'll be a bit clearer about that in the next release. `split_ci` delegates the CI calculations to the function `bcajack` in the `bcaboot` package, and returns whatever that function returns. This contains confidence intervals, as well as a host of other information. To find out what the function precisely returns, see https://rdrr.io/cran/bcaboot/man/bcajack.html

The coefficients are indeed in the bca column, so for the 0.025% lower limit of the CI, its value is 0.8243435

**Reply by tpronk** [reply]
Released v2.2.0, which features a fix to that `split_ci` issue you found. Give a shout when you've got some personal website (like ResearchGate or LinkedIn), then I'll update the acknowledgments to link to that site.

**Reply by SeraMaren** [reply]
Nice! Actually, yes my LinkedIn page would be nice. This is my LinkedIn:
https://www.linkedin.com/in/sera-maren-wiechert-b37625122/ 

Thanks!

**Reply by tpronk** [reply]
Coolio! I updated your name to Sera-Maren Wiechert and added links to LinkedIn. It's now in the source and README; at the next release it's also in the package documentation.
